runtime/list.go

```diff
@@ -338,6 +338,72 @@ func listNE(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return listCompare(f, toListUnsafe(v), w, NE)
 }
 
+func listIndex(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	expectedTypes := []*Type{ListType, ObjectType, ObjectType, ObjectType}
+	argc := len(args)
+	var raised *BaseException
+	if argc == 2 || argc == 3 {
+		expectedTypes = expectedTypes[:argc]
+	}
+	if raised = checkMethodArgs(f, "index", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	l := toListUnsafe(args[0])
+	var start, stop int
+	l.mutex.RLock()
+	switch argc {
+	case 2:
+		start = 0
+		stop = len(l.elems)
+	case 3:
+		numElems := len(l.elems)
+		start, raised = IndexInt(f, args[2])
```

Comment:
I think you can eliminate a bit of code duplication if you rearrange this as repeated if statements rather than a switch:

```
start, stop := 0, len(l.elems)
if argc >= 2 {
  start, raised = IndexInt(f, args[2])
  ...
}
if argc >= 3 {
  stop, raised = IndexInt(f, args[3])
  ...
}
```